lib/migrations/util/is-module-type.js

```diff
@@ -0,0 +1,16 @@
+const { readFile } = require('./fs');
+
+module.exports = async function isModuleType(filepath) {
+    try {
+        if (process.env.npm_package_json) {
+            // npm >= 7.0.0
+            const packageJson = JSON.parse(await readFile(process.env.npm_package_json, 'utf-8'));
+            if (packageJson.type === 'module') {
+                return true;
+            }
+        }
+    } catch (e) {
+        console.warn('Error reading package.json', e);
```

Comment:
In which scenarios do you expect this logic to fail?

Reply:
Now that I think about it it should not be something that can occur, as `npm` would refuse to run if `package.json` had an invalid syntax.